services/director-v2/src/simcore_service_director_v2/modules/celery.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from dataclasses import dataclass
+from typing import List
+
+from celery import Celery, Task
+from celery.contrib.abortable import AbortableAsyncResult
+from fastapi import FastAPI
+from models_library.projects import ProjectID
+from models_library.settings.celery import CeleryConfig
+
+from ..models.schemas.constants import UserID
+from ..utils.client_decorators import handle_retry
+
+logger = logging.getLogger(__name__)
+
+
+def setup(app: FastAPI, settings: CeleryConfig):
+    if not settings:
+        settings = CeleryConfig.create_from_env()
+
+    def on_startup() -> None:
+        CeleryClient.create(
+            app,
+            client=Celery(
+                settings.task_name,
+                broker=settings.broker_url,
+                backend=settings.result_backend,
+            ),
+            settings=settings,
+        )
+
+    async def on_shutdown() -> None:
+        del app.state.celery_client
+
+    app.add_event_handler("startup", on_startup)
+    app.add_event_handler("shutdown", on_shutdown)
+
+
+@dataclass
+class CeleryClient:
+    client: Celery
+    settings: CeleryConfig
+
+    @classmethod
+    def create(cls, app: FastAPI, *args, **kwargs):
+        app.state.celery_client = cls(*args, **kwargs)
+        return cls.instance(app)
+
+    @classmethod
+    def instance(cls, app: FastAPI):
+        return app.state.celery_client
+
+    @handle_retry(logger)
+    def send_task(self, task_name: str, *args, **kwargs) -> Task:
+        return self.client.send_task(task_name, *args, **kwargs)
+
+    def send_computation_task(self, user_id: UserID, project_id: ProjectID) -> Task:
+        return self.send_task(
+            self.settings.task_name,
+            expires=self.settings.publication_timeout,
+            kwargs={"user_id": user_id, "project_id": str(project_id)},
+        )
+
+    @classmethod
+    def abort_computation_tasks(cls, task_ids: List[str]) -> None:
+        for task_id in task_ids:
+            task_result = AbortableAsyncResult(task_id)
+            if task_result:
+                task_result.abort()
```

Comment:
apparently, according to the [doc](https://docs.celeryproject.org/en/2.4-archived/reference/celery.contrib.abortable.html?highlight=abortableasyncresult#celery.contrib.abortable.AbortableAsyncResult.abort), ``abort()`` only signals the task and does not guarantee that the task actually abort.  Did you consider that?
Should we add a check to ensure, i..e. something like 
```python
while not task_result.is_aborted(): 
       await async.sleep(.2) 
```
?

Reply:
so, abort uses the result_backend facilities. it sets the task as aborted.
then it is up to the worker to check if the task was aborted (which it hopefully does in the sidecar). once the sidecar effectively stops it sets the comp_task entry to ABORTED.
I am not sure it make sense to wait for the tasks to be aborted in that case. I do it when deleting the pipeline.